src/analysisd/rules.c

```diff
@@ -1404,11 +1404,12 @@ int Rules_OP_ReadRules(const char *rulefile)
                                 merror("Invalid option '%s' for "
                                 "rule '%d'", mitre_opt[ind]->element,
                                 config_ruleinfo->sigid);
-
-                                for (l = 0; config_ruleinfo->mitre_id[l] != NULL; l++) {
-                                    os_free(config_ruleinfo->mitre_id[l]);
+                                if (config_ruleinfo->mitre_id) {
```

Comment:
We could replace this block with:
```c
free_strarray(config_ruleinfo->mitre_id);
```

In fact, this function already checks that `mitre_id` is not null;